File: New Unity Project/Assets/nodeInteraction.js
#pragma strict


function Start () {

}

function Update () {
	
}

function OnMouseOver () {
	var gameObject = GameObject.FindGameObjectWithTag("GameController");
	var gameManager : init = gameObject.GetComponent("init");
	
	if (Input.GetMouseButton(1)) {
		var selectObj : squadMovement = gameManager.selected.GetComponent("squadMovement");
		selectObj.target = transform.position;

	}
}

File: Assets/NodeInteraction.cs
using UnityEngine;
using System.Collections;

public class NodeInteraction : MonoBehaviour {

	public GameManager GM;

	// Use this for initialization
	void Start () {
	
	}
	
	// Update is called once per frame
	void Update () {
	
	}

	void OnMouseOver () {	
		if (Input.GetMouseButton(1)) {
			SquadInteraction squad = GM.selected.GetComponent<SquadInteraction>(); 
			squad.target = transform.position;
		}
	}
}
